OpenSim/Common/Object.h

```diff
@@ -1086,50 +1085,85 @@ template <> struct Object_GetClassName<SimTK::Vector_<SimTK::Vec3>>
 #define OpenSim_OBJECT_TEMPLATE_DEFS(ConcreteClass, TArg, SuperClass)          \
+/** @name Auto-generated functions                                          */ \
+/** @{                                                                      */ \
+/** This returns "##ConcreteClass##_<T>_".                                  */ \
+/** T is the template argument for this class.                              */ \
+/** See getConcreteClassName() if you want the class name of the underlying */ \
+/** concrete object instead.                                                */ \
+/** \internal This is generated by the `OpenSim_DECLARE_*_OBJECT` macros.   */ \
 static const std::string& getClassName()                                       \
 {   static std::string name = #ConcreteClass "_"                               \
                               + Object_GetClassName<TArg>::name()              \
                               + "_";                                           \
-    return name; }
+    return name; }                                                             \
+/** @}*/
 
 // This provides definitions for the two Object pure virtuals clone() and
 // getConcreteClassName().
 #define OpenSim_OBJECT_CONCRETE_DEFS(ConcreteClass)                            \
+/** @name Auto-generated functions                                          */ \
+/** @{                                                                      */ \
 ConcreteClass* clone() const override {return new ConcreteClass(*this);}       \
 const std::string& getConcreteClassName() const override                       \
 {   return getClassName(); }                                                   \
+/** @}*/                                                                       \
 private:
 
 // This leaves the two Object pure virtuals clone() and getConcreteClassName()
 // unimplemented, but changes the return type of clone() to ConcreteClass*,
 // which allows it to be invoked ConcreteClass::clone() and return the correct
 // pointer type.
 #define OpenSim_OBJECT_ABSTRACT_DEFS(ConcreteClass)                            \
+/** @name Auto-generated functions                                          */ \
+/** @{                                                                      */ \
 ConcreteClass* clone() const override = 0;                                     \
 const std::string& getConcreteClassName() const override = 0;                  \
+/** @}*/                                                                       \
 private:
 
 // Add public static method declaration in class to assist in downcasting
 // arbitrary objects to the new type to support dynamic casting across JNI.
 #define OpenSim_OBJECT_JAVA_DEFS(thisClass) \
   public: \
+  /** @name Auto-generated functions */ \
+  /** @{ */ \
+  /** For use in MATLAB and Python to access the concrete class. */ \
   static thisClass* safeDownCast(OpenSim::Object *obj) \
   { \
       return dynamic_cast<thisClass *>(obj); \
```

Comment:
Do you think it'd help to say this's the equivalent of dynamic_cast<thisClass *> for Scripting languages? In fact it can also be used in C++ for people not comfortable with the cast syntax I think.